### ISSUE TITLE
Defining multiple thermistors

### ISSUE THREAD

**Issue by Moerup** [KevinOConnor/klipper#1189]
Would it be possible to add functionality to define extra thermistors that could be used for reading temperature around different aspects of the printer?

Currently possible with creating a "fake" extruder and assigning the thermistor to that, then it even shows up in Octoprint without any modifications.
Just seems like a wrong way to do it. 
Maybe something along the lines of this in the config file:
```
[thermistor temp_x]
sensor_type: NTC 100K beta 3950
sensor_pin: analog13
```
And then have a method like the QUERY_ENDSTOP to get the readings? Or something that updates constantly like the extruder/bed temps?

Atm trying to setup thermistors on my stepper drivers to monitor heat on them, as I'm having some overheating problems. This would be a huuuge help! Also for people that wants to control the chamber temp and stuff like that.

**Comment by wizhippo** [comment]
I to am interested in this but would like to be able to shut the board down off after a temp threshold.  For example I want to add a thermistor to my heat bed mossfet and shut off power if too hot.

**Comment by JamieAKing** [comment]
I would also like this.
Am I right in thinking that a pmw fan can be configured to operate using a thermistor plugged into a spare pin on the board, we just can’t see the value of that thermistor in octoprint?

**Comment by wizhippo** [comment]
Additionally would be nice it the responses were added to m105 show it shows in optoprints graphs

**Comment by wizhippo** [comment]
Actually rethinking this and temperature fan, both could be replaced by the generic_heater.

If you could setup a generic heater with a 'null' pwm device this would work.  The heater logic would show the temps, go to shutdown state etc.

Then maybe a few changes such as adding a min pwm and kickstart time to heater and heater verify you could also replace temperature fans as the really are  heater in reverse.

**Comment by wizhippo** [comment]
After testing using a generic_heater as a temperature fan works great.  Just have to invert the output and modify the heater verify

**Comment by wizhippo** [comment]
See https://github.com/KevinOConnor/klipper/pull/1258 example of mosfet monitoring only.

**Comment by nettunoroberto** [comment]
"Just have to invert the output and modify the heater verify" I have to modify files to do this?

**Comment by mental405** [comment]
Currently dabbling with a couple of thermistors over my stepper drivers and FETs. Would be great if there were a way to report them through the terminal.

**Comment by wizhippo** [comment]
I have added a pull request that should work for you.  You setup generic heaters that have `none` control.

https://github.com/KevinOConnor/klipper/pull/1268

**Comment by KevinOConnor** [comment]
FYI, support for arbitrary temperature sensors is now available (commit b6589406).

-Kevin
